### DISCUSSION TITLE
AuthApiError: Database error saving new user     at eval (fetch.js:49:20)

### DISCUSSION THREAD

**Discussion by shalawfatah** [supabase/supabase#12673] [Questions]
I want to submit a new user 

```
const supabase = useSupabaseClient()
const submitForm = async (e) => {
    e.preventDefault()
    const { data, error } = await supabase.auth.signUp({
        email: email,
        password: password,
        options: {
            data: {
              full_name: name,
              avatar: ''
            },
          },
      })
```

Somehow the database rejects it without much explanation. I think I have all `RLS` disabled.

**Comment by GaryAustin1** [comment]
You likely have an insert trigger on auth.users for generating a profile and it is not working.
You can see more about the error by using the dashboard postgres log right after the error.
Also if you have a trigger function you can temporarily turn off the trigger in the dashboard/database/trigger section to verify it is the issue.
RLS does not play a role in signUp, unless you have a trigger function trying to use other tables.

**Reply by shalawfatah** [reply]
Thanks @GaryAustin1 , I removed the SQL trigger completely, removed everything but this, but I still have the same issue: 

```
    const { data, error } = await supabase.auth.signUp({
        email: email,
        password: password
      })
```

**Reply by GaryAustin1** [reply]
Is there an error in the postgres logs from the dashboard right after you do the signUp?

**Reply by shalawfatah** [reply]
@GaryAustin1 Even after removing the SQL trigger and handler, it seems a connection remains, look at this: 
```
INSERT INTO auth.users (id, email, encrypted_password) VALUES (uuid_generate_v4(), 'johndoe@example.com', 'Password');

```
The error is: 
```
Failed to run sql query: column "avatar_url" of relation "users" does not exist

```

**Reply by shalawfatah** [reply]
This is another error:
```
Failed to run sql query: relation "public.profiles" does not exist
```
but I can't see any relations anywhere as I removed the profiles table.

**Reply by GaryAustin1** [reply]
Seems like you still have a trigger function being called.

**Reply by GaryAustin1** [reply]
Run this.   
```
SELECT  event_object_table AS table_name ,trigger_name         
FROM information_schema.triggers  
GROUP BY table_name , trigger_name 
ORDER BY table_name ,trigger_name 
```

**Reply by shalawfatah** [reply]
Yes, I see that the `handle_new_user` function inside functions in the dashboard still there, when I try to remove it, it says you can't remove it as other `objects` depend on it.

**Reply by GaryAustin1** [reply]
You have a trigger still applied most likely

**Reply by shalawfatah** [reply]
Thanks @GaryAustin1 I solved that problem, although I still cannot use `jsonb` for some reason even if I type cast, it gives me the error below: 

```
INSERT INTO auth.users (id, email, encrypted_password, raw_user_meta_data)
VALUES (
    uuid_generate_v4(), 
    'john@doe.com', 
    'Password', 
    '{
      "avatar": "https://images.pexels.com/photos/4355346/pexels-photo-4355346.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2",
      "full_name": "Ultra",
      "description": "New desc",
      "latlng": {
        "lat": 49.2,
        "lng": -122.9
      }
    }'::jsonb
);
```

It says:
```
Failed to run sql query: invalid input syntax for type json
```

**Reply by GaryAustin1** [reply]
No idea.  Your exact code works on my test sites auth.users table.
![image](https://user-images.githubusercontent.com/54564956/221382230-8c518ae0-a8ec-4835-a763-148f9fda10cb.png)
